File: 1548-check-if-all-1s-are-at-least-length-k-places-away/1548-check-if-all-1s-are-at-least-length-k-places-away.go
func kLengthApart(nums []int, k int) bool {
	last := -k - 1
	for curr := 0; curr < len(nums); curr++ {
		if nums[curr] == 1 {
			if curr-last <= k {
				return false
			}
			last = curr
		}
	}
	return true
}

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/1548-check-if-all-1s-are-at-least-length-k-places-away.rs
impl Solution {
    pub fn k_length_apart(nums: Vec<i32>, k: i32) -> bool {
        let mut last = -k as isize - 1;
        for curr in 0..nums.len() {
            if nums[curr] == 1 {
                if curr - last as usize <= k as usize {
                    return false;
                }
                last = curr as isize;
            }
        }

        true
    }
}